Converted user input to number multiple-choice options

soumyaray opened the pull request:
Changed the user input so that it presents players with a multiple-choice of committers that they can numerically select from.  Also, some other idiomatic changes.

Based on the suggestion of the [top comment on Reddit](http://www.reddit.com/r/programming/comments/2x7vde/the_git_game_guess_who_on_your_team_wrote_a/). 

Hope you like it!

soumyaray committed:
Converted user input to number multiple-choice options

andbroby commented:
Couple of issues here:
1. [This](https://github.com/jsomers/git-game/pull/23/files#diff-b3417ec47ee8a4a3f3b3de186ec61d39R98) matches 0, which would cause the guess to be the last option ([L104](https://github.com/soumyaray/git-game/blob/master/git-game#L104)). Should be `[1-9]+` if you want to use regex, or alternatively just rewrite the whole conditional using [between](http://ruby-doc.org//core-2.2.0/Comparable.html#method-i-between-3F).
2. It's pretty much unplayable in big repos since all the contributors from [L70](https://github.com/jsomers/git-game/pull/23/files#diff-b3417ec47ee8a4a3f3b3de186ec61d39R70) fill the screen :stuck_out_tongue:. Should be restricted to a set number of choices.

See how this guy did it: [PR #16](https://github.com/jsomers/git-game/pull/16).

soumyaray committed:
Multiple choices reduced to 5 at most for larger repos

- create_choices and get_multiple methods introduced
- limited choices used throughout instead of all committers
- using #between to check validity of input

soumyaray commented:
Thanks for the tips @andbroby:
1. Using `between` now to check range of input.  still using `/[0-9]+/` because `/[1-9]+/` would preclude `10` or `20`. This means that inputs like `03` are valid.
2. I was being lazy!  Changed it now so that 5 options are presented at random (always including proper author). I'm using `.sort` instead `.shuffle` so that the options are always in alphabetical order. For repos with 5 or less authors, the options will always be the same.

Let me know any other suggestions -- thanks again!

jsomers commented:
Thanks for the pull! I ended up going with @esjeon's https://github.com/jsomers/git-game/pull/16
